pallet-chainlink-feed/src/lib.rs

```diff
@@ -701,7 +701,8 @@ pub mod pallet {
 		/// Updates the pruning window of an existing feed
 		///
 		/// - Will prune rounds if the given window is smaller than the existing one.
-		#[pallet::weight(1_000)]
+		#[pallet::weight(T::WeightInfo::set_pruning_window(
+            <Feed<T>>::count_overflow_windows(*feed_id, *pruning_window)))]
```

Comment:
can we make this a one liner? 
```rust
Feed::<T>::load_from(feed_id).map(|feed|feed.current_window()).unwrap_or_default().saturating_sub(pruning_window)
            <Feed<T>>::count_overflow_windows(*feed_id, *pruning_window)))]
```
then we don't need the `count_overflow` function anymore.

Reply:
This is ugly LOL